rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -6551,9 +6551,123 @@ private void assertRefundInProcessPegInVersionLegacy(
         Assert.assertFalse(provider.getHeightIfBtcTxhashIsAlreadyProcessed(btcTx.getHash()).isPresent());
     }
 
+    private BigInteger registerFastBridgeBtcTransaction_RSKIP293(ActivationConfig.ForBlock activations, Coin valueToSend) throws IOException, BlockStoreException, BridgeIllegalArgumentException {
+
+        Context btcContext = mock(Context.class);
+        when(btcContext.getParams()).thenReturn(bridgeConstants.getBtcParams());
+
+        Repository repository = spy(createRepository());
+        BridgeStorageProvider provider = new BridgeStorageProvider(
+                repository,
+                PrecompiledContracts.BRIDGE_ADDR,
+                bridgeConstants,
+                activations
+        );
+
+        FederationSupport federationSupportMock = mock(FederationSupport.class);
+        doReturn(provider.getNewFederationBtcUTXOs()).when(federationSupportMock).getActiveFederationBtcUTXOs();
+
+        BridgeSupport bridgeSupport = spy(new BridgeSupport(
+                bridgeConstants,
+                provider,
+                mock(BridgeEventLogger.class),
+                new BtcLockSenderProvider(),
+                new PeginInstructionsProvider(),
+                repository,
+                mock(Block.class),
+                btcContext,
+                federationSupportMock,
+                mock(BtcBlockStoreWithCache.Factory.class),
+                activations
+        ));
+
+        Federation activeFederation = getFederation(bridgeConstants, "fa03", "fa04");
+        doReturn(activeFederation).when(bridgeSupport).getActiveFederation();
+
+        doReturn(bridgeConstants.getGenesisFederation()).when(bridgeSupport).getRetiringFederation();
+
+        doReturn(true).when(bridgeSupport).validationsForRegisterBtcTransaction(any(), anyInt(), any(), any());
+        doReturn(PegTestUtils.createHash3(1)).when(bridgeSupport).getFastBridgeDerivationHash(
+                any(Keccak256.class),
+                any(Address.class),
+                any(Address.class),
+                any(RskAddress.class)
+        );
+
+        Address btcAddress = Address.fromBase58(
+                btcParams,
+                "n3PLxDiwWqa5uH7fSbHCxS6VAjD9Y7Rwkj"
+        );
+
+
+        ECKey key = ECKey.fromPublicOnly(new BtcECKey().getPubKey());
+        RskAddress lbcAddress = new RskAddress(key.getAddress());
+
+
+        Address activeFederationAddress = getFastBridgeAddressFromRedeemScript(
+                activeFederation.getRedeemScript(),
+                1
+        );
+
+        Address retiringFederationAddress = getFastBridgeFederationAddress();
```

Comment:
This could become an error if someone changes the Federation used in `getFastBridgeFederationAddress`